### ISSUE TITLE
Rule Change: `array-callback-return` should support Stage 4 `Array.prototype.toSorted``

### ISSUE THREAD

**Issue by sosukesuzuki** [eslint/eslint#16842]
### What rule do you want to change?

`array-callback-return`

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
[1, 2, 3].toSorted(
  (a, b) {
    if (cond1) return -1
    if (cond2) return 1
  }
);
```

### What does the rule currently do for this code?

Do nothing

### What will the rule do after it's changed?

Warns as for `Array.prototype.sort`.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Change array by copy proposal reaches to Stage 4
ref:  https://github.com/tc39/proposals/commit/ad4df8435f27f39eda26db3b940ae151980c8015

**Comment by nzakas** [comment]
Makes sense to me. Feel free to open a pull request.
